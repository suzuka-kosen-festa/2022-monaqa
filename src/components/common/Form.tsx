import React from 'react'
import type { FC } from 'react'
import tw from 'twin.macro'

import { Textfield } from './Textfield'
import { Button } from './Button'

interface FormInterface {
  children: React.ReactNode
  buttonText: string
}

const FormBox = tw.div`flex flex-col space-y-2 justify-center  items-center w-full lg:(flex-row space-x-2 space-y-0 w-1/2)  `
const Label = tw.label`block text-gray-500 font-bold  text-lg text-right lg:(text-xl)  `
const Damy = () => null

const Form: FC<FormInterface> = ({ children, buttonText }) => (
  <FormBox>
    <Label>{children}</Label>
    <Textfield type="search" name="search" maxLength={20} />
    <Button buttonKey="accept" onClick={() => Damy}>
      {`${buttonText}`}
    </Button>
  </FormBox>
)

export { Form }
