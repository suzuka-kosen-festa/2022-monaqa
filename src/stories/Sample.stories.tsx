import type { ComponentStoryObj, ComponentMeta } from '@storybook/react'
import { Sample } from '../components/Sample'

type T = typeof Sample
type Story = ComponentStoryObj<T>

export default {
  component: Sample,
} as ComponentMeta<T>

export const Default: Story = {}
