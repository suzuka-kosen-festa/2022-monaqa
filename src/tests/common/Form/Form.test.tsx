import { composeStories } from '@storybook/testing-react'
import '@testing-library/jest-dom'
import { render, screen, fireEvent } from '@testing-library/react'
import React from 'react'
import * as stories from '../../../stories/common/Form.stories'

const { Default } = composeStories(stories)

describe('(components/common) Form', () => {
  test('take snap shot', () => {
    const { container } = render(<Default />)
    expect(container).toMatchSnapshot()
  })
})

describe('(components/common) Form', () => {
  test('input test', async () => {
    render(<Default />)
    fireEvent.change(screen.getByRole('searchbox'), {
      target: { value: "';delete from user_table;" },
    })
    expect(
      await screen.getByDisplayValue('&#x27;;delete from user_table;'),
    ).toBeInTheDocument()
  })
})
