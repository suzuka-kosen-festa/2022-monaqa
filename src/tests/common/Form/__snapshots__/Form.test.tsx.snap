// Vitest Snapshot v1

exports[`(components/common) Form > take snap shot 1`] = `
<div>
  <div
    class="css-1e3hdvz"
  >
    <label
      class="css-pkyrm3"
    >
      This is test
    </label>
    <input
      class="css-1f8go5f"
      name="search"
      type="search"
      value=""
    />
    <button
      class="css-177jtcm-Button"
    >
      <p
        class="css-7rlnkq-Button"
      >
        This is test
      </p>
    </button>
  </div>
</div>
`;
